server/src/main/java/org/jetbrains/bsp/bazel/server/sync/JUnitTestParser.kt

```diff
@@ -0,0 +1,163 @@
+package org.jetbrains.bsp.bazel.server.sync
+// TODO - rethink the location
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+import ch.epfl.scala.bsp4j.TaskId
+import ch.epfl.scala.bsp4j.TestReport
+import ch.epfl.scala.bsp4j.TestStatus
+import org.jetbrains.bsp.bazel.bazelrunner.BazelProcessResult
+import org.jetbrains.bsp.bazel.logger.BspClientTestNotifier
+import java.util.*
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+private data class TestOutputLine(
+  val name: String,
+  val passed: Boolean,
+  val message: String,
+  val indent: Int,
+  val taskId: TaskId?
+)
+
+private data class StartedBuildTarget(
+  val uri: String,
+  val taskId: TaskId
+)
+
+class JUnitTestParser(
+  private val bspClientTestNotifier: BspClientTestNotifier
+) {
+  fun processTestOutputWithJUnit(testResult: BazelProcessResult) {
+    val startedSuites: Stack<TestOutputLine> = Stack()
+    var startedBuildTarget: StartedBuildTarget? = null
+    var previousOutputLine: TestOutputLine? = null
+
+    testResult.stdoutLines.forEach {
+      if (startedBuildTarget == null) {
+        startedBuildTarget = checkLineForTestingBeginning(it)
+      } else {
+        val currentLineMatcher = testLinePattern.matcher(it)
```

Comment:
this matcher could be moved to the `getCurrentOutputLine`

Reply:
moved